src/main/java/org/mule/tools/devkit/sonar/checks/pom/DistributionManagementByCategoryCheck.java

```diff
@@ -0,0 +1,86 @@
+package org.mule.tools.devkit.sonar.checks.pom;
+
+import com.google.common.collect.Lists;
+import org.apache.maven.model.DeploymentRepository;
+import org.apache.maven.model.DistributionManagement;
+import org.apache.maven.project.MavenProject;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.slf4j.LoggerFactory;
+import org.sonar.check.Rule;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+
+import java.util.List;
+
+@Rule(key = DistributionManagementByCategoryCheck.KEY, name = "Distribution Management must be properly configured in pom.xml", description = "Checks that <distributionManagement> is declared in pom.xml and correctly configured (both <repository> and <snapshotRepository>) based on the connector category.", tags = { "connector-certification" })
+@ActivatedByDefault
+public class DistributionManagementByCategoryCheck implements PomCheck {
```

Comment:
This class needs to be added ConnectorsChecks.pomChecks() and MavenSensor.buildPomAllChecks().